internal/controller/kuadrant_status_updater.go

```diff
@@ -145,6 +162,117 @@ func mtlsLimitador(kObj *kuadrantv1beta1.Kuadrant, state *sync.Map) *bool {
 	return ptr.To(kObj.IsMTLSLimitadorEnabled() && len(effectiveRateLimitPoliciesMap) > 0)
 }
 
+func (r *KuadrantStatusUpdater) resilienceCondition(topology *machinery.Topology) ([]*metav1.Condition, []string) {
+	create := make([]*metav1.Condition, 0)
+	remove := make([]string, 0)
+
+	kObj := GetKuadrantFromTopology(topology)
+	isConfigured := kObj.Spec.Resilience.IsRateLimitingConfigured()
+	if !isConfigured {
+		remove = append(remove, ResilienceInfoRRConditionType, ResilienceWarningRRConditionType, ResilienceInfoPDBConditionType, ResilienceInfoTSCConditionType)
+		return nil, remove
+	}
+
+	lObj := GetLimitadorFromTopology(topology)
+	if lObj == nil {
+		remove = append(remove, ResilienceInfoRRConditionType, ResilienceWarningRRConditionType, ResilienceInfoPDBConditionType, ResilienceInfoTSCConditionType)
+		return nil, remove
+	}
+
+	if lObj.Spec.Replicas != nil && *lObj.Spec.Replicas < LimitadorReplicas {
+		cond := &metav1.Condition{
+			Type:    ResilienceWarningRRConditionType,
+			Message: fmt.Sprintf("Number of Limitador replicas (%v) below minimum default", *lObj.Spec.Replicas),
+			Reason:  "UserModifiedLimitadorReplicas",
+			Status:  metav1.ConditionUnknown,
+		}
+		create = append(create, cond)
+	} else {
+		remove = append(remove, ResilienceWarningRRConditionType)
+	}
+
+	if lObj.Spec.Replicas != nil && *lObj.Spec.Replicas > LimitadorReplicas {
+		cond := &metav1.Condition{
+			Type:    ResilienceInfoRRConditionType,
+			Message: fmt.Sprintf("Number of Limitador replicas (%v) greater than minimum default", *lObj.Spec.Replicas),
+			Reason:  "UserModifiedLimitadorReplicas",
+			Status:  metav1.ConditionUnknown,
+		}
+		create = append(create, cond)
+	} else {
+		remove = append(remove, ResilienceInfoRRConditionType)
+	}
+
+	if lObj.Spec.PodDisruptionBudget != nil && lObj.Spec.PodDisruptionBudget.MaxUnavailable != nil && *&lObj.Spec.PodDisruptionBudget.MaxUnavailable.IntVal != LimitadorPDB {
+		cond := &metav1.Condition{
+			Type:    ResilienceInfoPDBConditionType,
+			Message: "Limitador recource Pod Disruption Budget differs from default configuration",
+			Reason:  "UserModifiedLimitadorPodDisruptionBudget",
+			Status:  metav1.ConditionUnknown,
+		}
+		create = append(create, cond)
+	} else if lObj.Spec.PodDisruptionBudget != nil && lObj.Spec.PodDisruptionBudget.MinAvailable != nil {
+		cond := &metav1.Condition{
+			Type:    ResilienceInfoPDBConditionType,
+			Message: "Limitador recource Pod Disruption Budget differs from default configuration",
+			Reason:  "UserModifiedLimitadorPodDisruptionBudget",
+			Status:  metav1.ConditionUnknown,
+		}
+		create = append(create, cond)
+	} else {
+		remove = append(remove, ResilienceInfoPDBConditionType)
+	}
+
+	// Get the status of the topology spread constraints.
+	deployment := GetDeploymentForParent(topology, kuadrantv1beta1.LimitadorGroupKind)
+	if deployment != nil {
+		message := make([]string, 0)
+		for _, item := range deployment.Spec.Template.Spec.TopologySpreadConstraints {
```

Comment:
Should we check if the length of topology constraints matches 2?

Reply:
Are the only allowed values for the topology spread constraints "kubernetes.io/hostname" and "kubernetes.io/zone"? If custom topology keys are allowed, we should not assume the list will only have a length of 2.